src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1021,16 +1021,8 @@ bool PolicyHandler::SendMessageToSDK(const BinaryMessage& pt_string,
   LOG4CXX_AUTO_TRACE(logger_);
   POLICY_LIB_CHECK(false);
 
-  ApplicationSharedPtr app;
-  uint32_t app_id = 0;
-  if (last_used_app_ids_.empty()) {
-    LOG4CXX_WARN(logger_, "last_used_app_ids_ is empty");
-    return false;
-  } else {
-    app_id = last_used_app_ids_.back();
-
-    app = application_manager_.application(app_id);
-  }
+  const uint32_t app_id = GetAppIdForSending();
+  ApplicationSharedPtr app = application_manager_.application(app_id);
```

Comment:
@AKalinich-Luxoft @dev-gh @okozlovlux This appears to just reverse changes from https://github.com/smartdevicelink/sdl_core/commit/6a3c65f64fb7e9e5a770476b3f664d57606af33a#diff-7ec0ea8e1e752c736fb05b5f89890fbfL970. Is there any reason that these changes were made in the first place, and are they necessary for policy modes other than `HTTP`?